assets/app/view/game/scrap_trains.rb

```diff
@@ -6,14 +6,36 @@ module View
   module Game
     class ScrapTrains < Snabberb::Component
       include Actionable
+      needs :corporation, default: nil
 
       def render
-        @corporation = @game.round.active_step.current_entity
-        step = @game.round.active_step
+        @corporation ||= @game.round.active_step.current_entity
+        step = @game.round.step_for(@corporation, 'scrap_train')
 
         scrappable_trains = step.scrappable_trains(@corporation)
         return nil if scrappable_trains.empty?
 
+        if step.scrap_trains_button_only?
+          render_button(scrappable_trains)
+        else
+          render_section(scrappable_trains)
+        end
+      end
+
+      def render_button(scrappable_trains)
+        step = @game.round.step_for(@corporation, 'scrap_train')
```

Comment:
This section is a duplicate of line 53 and on. Can you break out into a method both call?

Reply:
I was able to do it